src/test_l1_dsHdmiIn.c

```diff
@@ -2348,20 +2355,47 @@ static UT_test_suite_t * pSuite2 = NULL;
  */
 int test_l1_dsHdmiIn_register ( void )
 {
-	/* add a suite to the registry */
-	pSuite = UT_add_suite( "[L1 dsHdmiIn]", NULL, NULL );
-    pSuite2 = UT_add_suite( "[L1 dsHdmiIn --  advanced]", NULL, NULL );
-	if ( NULL == pSuite )
-	{
-		return -1;
-	}	
+        ut_kvp_status_t status;
+        int32_t source_type = 0;
+	pSuite2 = UT_add_suite( "[L1 dsHdmiIn --  advanced]", NULL, NULL );
+        
+	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "dsHdmiIn.Type", gDeviceType, TEST_DS_DEVICE_TYPE_SIZE);
+
+	if (status == UT_KVP_STATUS_SUCCESS ) {
+        if (!strncmp(gDeviceType, TEST_TYPE_SOURCE_VALUE, TEST_DS_DEVICE_TYPE_SIZE)) {
```

Comment:
Align the code with proper spacing. Also, ensure the compilation and execution parts are functioning correctly. Verify that both the source and sink test cases listing properly when passing the relevant YAML profile file.

Reply:
Done

Reply:
Done